src/extra/others/controller/lv_obj_controller.h

```diff
@@ -0,0 +1,216 @@
+    void (*obj_deleted_cb)(struct lv_obj_controller_t * self, lv_obj_t * obj);
+
+    /**
+     * Handle event
+     * @param self Controller instance
+     * @param which User-defined ID of event
+     * @param data1 User-defined data
+     * @param data2 User-defined data
+     */
+    bool (*event_cb)(struct lv_obj_controller_t * self, int which, void * data1, void * data2);
+
+    /**
+     * *REQUIRED*: Allocation size of controller
+     */
+    size_t instance_size;
+};
+
+/**********************
+ * GLOBAL PROTOTYPES
+ **********************/
+
+/**
+ * Create controller manager instance
+ * @param container Container object for manager to add objects to
+ * @param parent Parent obj_controller if this manager is nested
+ * @return Controller manager instance
+ */
+lv_controller_manager_t * lv_controller_manager_create(lv_obj_t * container, lv_obj_controller_t * parent);
+
+/**
+ * Destroy controller manager instance
+ * @param manager Controller manager instance
+ */
+void lv_controller_manager_del(lv_controller_manager_t * manager);
+
+/**
+ * Remove the top-most controller for stack
+ * @param manager Controller manager instance
+ */
+void lv_controller_manager_pop(lv_controller_manager_t * manager);
+
+/**
+ * Add controller to the stack
+ * @param manager Controller manager instance
+ * @param cls Controller class
+ * @args User-defined argument
+ */
+void lv_controller_manager_push(lv_controller_manager_t * manager, const lv_obj_controller_class_t * cls, void * args);
+
+/**
+ * Replace top-most controller. Old item in the stack will be removed.
+ * @param manager
+ * @param cls
+ * @param args
+ */
+void lv_controller_manager_replace(lv_controller_manager_t * manager, const lv_obj_controller_class_t * cls,
+                                   void * args);
+
+/**
+ * Get stack size of this controller manager
+ * @param manager Controller manager instance
+ * @return Stack size of this controller manager
+ */
+size_t lv_controller_manager_size(lv_controller_manager_t * manager);
+
+/**
+ * Send event to top-most controller
+ * @param manager Controller manager instance
+ * @param which User-defined ID of event
+ * @param data1 User-defined data
+ * @param data2 User-defined data
+ * @return true if controller returned true
+ */
+bool lv_controller_manager_dispatch_event(lv_controller_manager_t * manager, int which, void * data1, void * data2);
+
+/**
+ * Show lv_msgbox
+ * @param manager Controller manager instance
+ * @param cls Controller class which must return valid lv_msgbox instance
+ * @param args
+ */
+void lv_controller_manager_show(lv_controller_manager_t * manager, const lv_obj_controller_class_t * cls, void * args);
+
+lv_obj_controller_t * lv_controller_manager_top_controller(lv_controller_manager_t * manager);
+
+lv_obj_controller_t * lv_controller_manager_parent(lv_controller_manager_t * manager);
+
+lv_obj_controller_t * lv_obj_controller_class_create_unmanaged(const lv_obj_controller_class_t * cls, lv_obj_t * parent,
```

Comment:
It's not clear to me what `unmanaged` exactly mean. Could you add more details to the docs to describe as much features as possible?

Reply:
I have added some descriptions.